### PR TITLE
Reorganize providers

### PR DESCRIPTION
**Pull request by speller26**
Also unified Braket Rydberg transpilers

### CODE REVIEW

**Comment by PicksPeng** [comment]
I have a question about the current structure. I assume that the current goal is to have providers as the sub-package and to put all files related to the devices accessible by the provider into the corresponding sub-package.

However, a backend device may be accessible by multiple providers: i.e., IonQ devices may be accessed through IonQ Quantum Cloud and Braket. Some utilities may be shared, like circuit optimizers. Do we want to maintain them separately for different providers?

A possible solution is to have three independent sub-packages: `aais`, `backends`, `providers`, and under `providers` we have sub-packages for each provider. But this may also introduce some confusion: i.e., what files should be maintained in `backends`.

Any suggestions?

File: simuq/backends/ionq_circuit.py
Comment (speller26) [inline simuq/backends/ionq_circuit.py:1]: Ignore this file; it was included because of a bad rebase

File: simuq/transpiler.py
Comment (licedric) [inline simuq/transpiler.py:1]: Any thoughts on whether this class is needed, given that the `transpile` method signatures may differ?

Comment (speller26) [reply simuq/transpiler.py:1]: The inputs are not the same but similar, so I'm going to try unifying them in another PR; if that doesn't work well, I'll remove the class.

**Pull request merged**